### PR TITLE
chore(ci): fix CI warnings

### PR DESCRIPTION
**Pull request by raducristianpopa**
## Changes proposed in this pull request

- Update workflows to the new `GITHUB_STATE` and `GITHUB_OUTPUT` environment files

## Context

- https://github.com/interledger/rafiki/issues/825
## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

**Comment by raducristianpopa** [comment]
Some of the GH Actions are still using `set-output` and `save-state` (ie: https://github.com/interledger/rafiki/actions/runs/3632845686).
Should we bump their versions?

By bumping their versions we will probably solve this too:
> Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

**Comment by wilsonianb** [comment]
>Should we bump their versions?

Yeah, I let's do that here to fully resolve the open issue. Thanks!

**Comment by raducristianpopa** [comment]
New warning showed up:
> Build / Release Auth Docker Image: .github#L1
ubuntu-latest workflows will use ubuntu-22.04 soon. For more details, see https://github.com/actions/runner-images/issues/6399

Is the Ubuntu version a concern in our case ?

**Comment by mkurapov** [comment]
I don't _think_ the `ubuntu-latest` warning applies since we are using [actions/setup-node](https://github.com/actions/setup-node) already

reference: https://github.com/actions/runner-images/issues/6399

File: .github/workflows/lint_test_build.yml [hunk 1]
```diff
@@ -42,8 +42,8 @@ jobs:
     needs: checkout
     timeout-minutes: 5
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
```
Comment (mkurapov) [inline .github/workflows/lint_test_build.yml:46]: I'm not 100% sure, but are these two lines even necessary if we are doing:
`uses: ./.github/workflows/rafiki/env-setup` where the workflow already does 

```yaml
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
```
?

Comment (wilsonianb) [reply .github/workflows/lint_test_build.yml:46]: It seems like they can be removed throughout this file wherever `env-setup` is used

**Comment by wilsonianb** [comment]
I opened an issue for the ubuntu version
- https://github.com/interledger/rafiki/issues/830

**Comment by raducristianpopa** [comment]
It looks like the "unnecessary steps", might be necessary? 🤔
https://github.com/interledger/rafiki/actions/runs/3655579411/jobs/6177061467#step:2:1

Reference: https://github.com/orgs/community/discussions/26245#discussioncomment-3250935

**Comment by wilsonianb** [comment]
:thinking: 
Seems like we could instead remove `actions/checkout` from `env-setup`
but make `actions/setup-node` only happen in `env-setup`?

**Comment by raducristianpopa** [comment]
> 🤔 Seems like we could instead remove `actions/checkout` from `env-setup` but make `actions/setup-node` only happen in `env-setup`?

All CI checks are passing with these changes.

**Review by mkurapov — APPROVED** [review]

**Review by wilsonianb — APPROVED** [review]
:tada:

**Pull request merged**